llvm/test/tools/llvm-objdump/wasm/linked-symbol-table-namesec.yaml

```diff
@@ -0,0 +1,87 @@
+# RUN: yaml2obj %s -o %t.wasm
+# RUN: llvm-objdump -t %t.wasm | FileCheck %s
```

Comment:
Should we add a test for `nm` too?

Reply:
Maybe? Either objdump or nm is sufficient to cover this code, since they both just dump info from the Object, and (hopefully) we have other tests for nm-specific things. I was thinking about updating nm to show symbol sizes for functions (currently we just do it for data symbols) so we could use a name section to test that and get some overlapping coverage.

Reply:
I added an nm test to also test that the linking section overrides the name section. I'll move both of the tests to test/Object after #81072 lands